### ISSUE TITLE
synctex unable to open

### ISSUE THREAD

**Issue by asiloisad** [allefeld/atom-pdfjs-viewer#6]
Hi,
when i try to use synctex via right mouse button then warning occur:
![image](https://user-images.githubusercontent.com/29706474/94761997-ce911080-03a6-11eb-9f80-d724077e52a8.png)
as result nothing happen (.tex file is not opened).
synctex is avaiable in system paths (in package pdf-view-plus working well).
![image](https://user-images.githubusercontent.com/29706474/94762146-35162e80-03a7-11eb-99da-f640ba8526bc.png)
windows 10 x64

edit: probably the quetes are needed:

    synctex edit -o 3:263:58:"C:\data\[p] samples\[b20201001] texme-check\tex\main.pdf"

Best regards!
P.S. best pdf viewer in atom 👍

**Comment by allefeld** [comment]
Thanks for the detailed info, that's helpful.

As far as I can tell, it is not a problem with the synctex call. Otherwise there would have been a message "PdfjsViewerView: SyncTeX failed with code XX" after the first message, or a notification "Could not run SyncTeX". Quotes are only needed when the command is given from the terminal/shell (or what it is called on Windows); here the synctex process is directly spawned with the arguments `edit`, `-o`, and `3:263:58:C:\data\[p] samples\[b20201001] texme-check\tex\main.pdf`, so the spaces shouldn't be an issue.

I think the error occurs when my package tries to open the tex file at the position returned by synctex. That's what the second message in the developer tools indicates, admittedly not very informatively.

The problem is, I can't directly debug the problem. I created a file with a path similar to yours, and for me the reverse lookup works: I get to the correct line in the tex file, which is either focused or opened. But I'm on Linux.

What happens under the hood is a call

    atom.workspace.open('C:\data\[p] samples\[b20201001] texme-check\tex\main.pdf')

plus some other options for the correct line etc. This first argument is called a URI in the [documentation](https://flight-manual.atom.io/api/v1.18.0/Workspace/#instance-open). Right now my only guess is that the backslashes need to be escaped, or URL-encoded, or replaced by forward slashes.

Could you go into the developer console and try each of the following commands, and see whether they successfully open the tex file?

````
atom.workspace.open('C:\data\[p] samples\[b20201001] texme-check\tex\main.pdf')
atom.workspace.open(encodeURI('C:\data\[p] samples\[b20201001] texme-check\tex\main.pdf'))
atom.workspace.open('C:/data/[p] samples/[b20201001] texme-check/tex/main.pdf')
````

PS: I doubt `encodeURI` is correct since it breaks things on Linux.

**Comment by asiloisad** [comment]
![image](https://user-images.githubusercontent.com/29706474/94859801-6c332100-0435-11eb-9511-dee0df54be4a.png)

only path with slash working well, more details at screenshot from console

@ edit: please remember, in windows 10 the slash can be used instead of backslash

**Comment by allefeld** [comment]
Thanks!

OK, it looks like I could simply replace backslashes by slashes in the path returned by SyncTeX.

It's just that I was unable to find any info on whether that's necessary or recommended in JavaScript / Electron / Atom / Node. I've posted a question on stackoverflow to check what's going on.

**Comment by asiloisad** [comment]
any updates on going?

**Comment by allefeld** [comment]
My questions on stackoverflow and codidact didn't clear the matter up, so I'm in the uncomfortable position of guessing...

I have made a commit ba4ce37b1d39a95ebf7f452365708e5674399a18 which replaces backslashes by slashes in the pathname returned by SyncTeX, only on Windows. Could you test it? Thanks!

I haven't *published* the new version though, so you would have to install the current state directly from GitHub.

-   Only the two files in the subfolder [lib](https://github.com/allefeld/atom-pdfjs-viewer/tree/master/lib) are affected, so you could simply download them and copy them over the files you have installed, i.e. into `[Atom Config]\packages\pdfjs-viewer\lib`.

-   Alternatively, running `apm install git@github.com:allefeld/atom-pdfjs-viewer.git` should work, too.

Let me know whether it works, then I'll publish the new version.

**Comment by allefeld** [comment]
@asiloisad, did you try out the changed version?

**Comment by manu-schaaf** [comment]
I'm having the same issue and I believe it to be caused by the windows line break: `\r\n`. I replaced the `output.split("\n")` in `pdfjs-viewer-view.js` line 205 with a Windows-friendly `output.split(/\r?\n/g)` (see [my fork](https://github.com/m-stoeckel/atom-pdfjs-viewer/blob/0eeb20fb2a23681b1db98c72d642f01e49a34c0e/lib/pdfjs-viewer-view.js#L205)) which fixes the problem in my case.

But I am genuinely surprised, that the quotes are not needed, because you **do** need them in the windows console.

**Comment by allefeld** [comment]
Hi @m-stoeckel, thanks for contributing! So you're saying replacing backslashes by slashes as I did in my provisional fix doesn't help, and the problem is something else entirely? I'm confused, but since you have a working fix, I guess you're right. But your fork also includes my provisional fix, so maybe it is necessary? It would be great if you could investigate that.

Yes, you need the quotes in the console, also in the shell on un*x. The purpose of the quotes is to designate the included as a single argument, prevent it from being broken up into several arguments. The difference is that the command started from my code does not go through the console/shell, it directly specifies the arguments. See
https://github.com/allefeld/atom-pdfjs-viewer/blob/ba4ce37b1d39a95ebf7f452365708e5674399a18/lib/pdfjs-viewer-view.js#L187
The command that appears in the JavaScript console is just a simple concatenation
https://github.com/allefeld/atom-pdfjs-viewer/blob/ba4ce37b1d39a95ebf7f452365708e5674399a18/lib/pdfjs-viewer-view.js#L199
It does not reflect what is actually going on.

**Comment by francois-gu** [comment]
Having the same issue. I tried @m-stoeckel fix and it is working fine so far.

By the way, thanks for the package @allefeld, very happy to be able to switch from overleaf to atom

**Comment by allefeld** [comment]
@Asphod, good to know, and glad you like the package.

Could you test whether the problem stays fixed when you remove the backslash-fixing lines (209–212)?

        if (process.platform == 'win32') {
          // attempt to fix issue #6
          synctex.input = synctex.input.replace('\\', '/')
        }

**Comment by hcsoso** [comment]
Having the same issue; confirming that @m-stoeckel's fix works without the backslash-fixing lines.

**Comment by allefeld** [comment]
Thanks @hcsoso!

**Comment by allefeld** [comment]
I included @m-stoeckel's fix and commented out my previous attempt. I assume that this should fix this issue.

Thanks everyone for helping and sorry it took me so long!

**Comment by allefeld** [comment]
published with v1.3.0
